hack/go-fmt.sh

```diff
@@ -5,7 +5,11 @@ if [ "$IS_CONTAINER" != "" ]; then
   done
   git diff --exit-code
 else
-  podman run --rm \
+  ENGINE="podman"
+  if [ "$(uname)" = "Darwin" ]; then
```

Comment:
Should we add this in if there is a work around? Could the user 1) alias podman=docker or 2) if we are going to follow through with these changes, warn someone as to the changes. 

```if ! command -v podman >/dev/null 2>&1 ; then
    if command -v docker >/dev/null 2>&1 ; then
        echo "[WARNING]: Aliasing docker to podman. Install podman if your system meets requirements.";
        alias podman=docker;
    else
        echo "[ERROR]: Failed to find services in the following order [podman, docker]." && exit
    fi
fi
```
It doesn't have to be this solution, but it would at least warn the user.